### DISCUSSION TITLE
Query HDFS in command line without any output (and command line hang)

### DISCUSSION THREAD

**Discussion by haowdev** [apache/iotdb#3836] [Q&A] [answered]
Experts,
I configure IOTDB with HDFS with very few data (less than 10 lines)
It seems the write to HDFS is OK, but read will timeout with following command line
![image](https://user-images.githubusercontent.com/40750896/130906140-9b708d52-69a3-4858-9552-2297720b1d7c.png)

**Comment by haowdev** [comment]
Any suggestion in how to trouble shoot this issue?
My Hadoop is a single node cluster v3.3.1.

**Reply by samperson1997** [reply]
Hi, may I ask which version of IoTDB are you using and have a look at your HDFS related configuration?

**Reply by haowdev** [reply]
My version is 0.12.1. I saw your version is 0.13.
Is that some fix in 0.13?

**Reply by samperson1997** [reply]
Actually not… this module hasn’t been updated for a while… I’ll try 0.12.1 again : ) (Or maybe you could try 0.13 which is master branch)

**Reply by haowdev** [reply]
Sure, thanks!

**Comment by samperson1997** [comment]
Hi, I've tried the latest version of IoTDB and I got the right results. Here is an example of HDFS configuration, and maybe you could check yours:
```
# TSFile storage file system. Currently, Tsfiles are supported to be stored in LOCAL file system or HDFS.
# Datatype: FSType
tsfile_storage_fs=HDFS

# If using HDFS, the absolute file path of Hadoop core-site.xml should be configured
# Datatype: String
core_site_path=/.../hadoop/etc/hadoop/core-site.xml

# If using HDFS, the absolute file path of Hadoop hdfs-site.xml should be configured
# Datatype: String
hdfs_site_path=/.../hadoop/etc/hadoop/hdfs-site.xml

# If using HDFS, hadoop ip can be configured. If there are more than one hdfs_ip, Hadoop HA is used
# Datatype: String
hdfs_ip=localhost

# If using HDFS, hadoop port can be configured
# Datatype: String
hdfs_port=9000
```
And here is my screen snapshot:
<img width="566" alt="1" src="https://user-images.githubusercontent.com/19167280/131512005-21534a5f-f205-41ea-8812-aaeda6518b74.png">

**Reply by haowdev** [reply]
Would you please to share the core-site.xml and hdfs-site.xml if possible?

**Reply by samperson1997** [reply]
Sure, here is mine as a simple example:

hdfs-site.xml:
```
<configuration>
	<property>
        <name>dfs.replication</name>
        <value>1</value>
    </property>
　　 <property>
        <name>dfs.namenode.name.dir</name>
        <value>/.../hadoop/tmp/hdfs/name</value>
    </property>
    <property>
        <name>dfs.datanode.data.dir</name>
        <value>/.../hadoop/tmp/hdfs/data</value>
    </property>
    <property>
        <name>dfs.webhdfs.enabled</name>
        <value>true</value>
    </property>
    <property>
        <name>dfs.http.address</name>
        <value>0.0.0.0:50070</value>
    </property>
</configuration>
```

core-site.xml:
```
<configuration>
	<property>
        <name>fs.defaultFS</name>
        <value>hdfs://0.0.0.0:9000</value>
    </property>
    <property>
        <name>hadoop.tmp.dir</name>
        <value>/.../hadoop/temp</value>
    </property>
</configuration>
```

**Comment by haowdev** [comment, marked as answer]
We can close the solution. I thing I met is that I downloaded source code of 0.12.1 then compile on a windows 10 but the compile failed. Then I downloaded the master branch to compile again on the same environment (it was success). I copied the hadoop-tsfile-***.jar to the lib folder then run the iodb server. That is how the issue I reported happens.
Today I found a Linux environment then build jar from source 0.12.1 on it.  After copy the hadoop-tsfile-***.jar to lib, the issue disappear.

**Reply by samperson1997** [reply]
Great! Thanks for your detailed explanation : )
